shared/rangeanalysis/codeql/rangeanalysis/RangeAnalysis.qll

```diff
@@ -194,18 +239,40 @@ signature module Semantic {
   /** Gets the type of an expression. */
   Type getExprType(Expr e);
 
+  /**
+   * A single static single assignment (SSA) variable.
+   */
   class SsaVariable {
+    /**
+     * Gets the expression where this SSA variable is used.
+     */
     Expr getAUse();
 
+    /**
+     * Gets the basic block where this SSA variable is defined.
+     */
     BasicBlock getBasicBlock();
   }
 
+  /**
+   * A phi node in the SSA form. A phi node is a kind of node in the SSA form
+   * that represents a merge point where multiple control flow paths converge
+   * and the value of a variable needs to be selected.
+   */
   class SsaPhiNode extends SsaVariable {
     /** Holds if `inp` is an input to the phi node along the edge originating in `bb`. */
     predicate hasInputFromBlock(SsaVariable inp, BasicBlock bb);
   }
 
+  /**
+   * A single update to a variable in the SSA form.
+   */
```

Comment:
I'd like to know what this class is for.

Reply:
It most commonly represents explicit assignments to variables, so that's fairly important to range analysis.

Reply:
(this QLDoc hsa been expanded a little since; further clarity can be gained from context, i.e. the other classes extending `SsaVariable`.  In any case, I'm marking this point as resolved now)